backend/pkg/auth/auth_config.go

```diff
@@ -9,9 +10,14 @@ type AuthConfig struct {
 	Password string
 }
 
-func NewAuthConfig(datasource *dbstore.SQLiteDatasource) *AuthConfig {
-	settings := datasource.GetSettings()
-	return &AuthConfig{Username: settings.GrafanaUsername, Password: settings.GrafanaPassword}
+func NewAuthConfig(datasource *dbstore.SQLiteDatasource) (*AuthConfig, error) {
+	settings, err := datasource.GetSettings()
+	if err != nil {
+		log.DefaultLogger.Error("NewAuthConfig: datasource.GetSettings(): ", err.Error())
+		return nil, err
+	}
+
+	return &AuthConfig{Username: settings.GrafanaUsername, Password: settings.GrafanaPassword}, nil
```

Comment:
Nice! Since this is a constructor, I'm not sure if returning an error is conventional or not, but it works!